ethdb/bitmapdb/dbutils.go

```diff
@@ -130,18 +161,17 @@ func TruncateRange(c ethdb.Cursor, key []byte, from, to uint64) error {
 		if err != nil {
 			return err
 		}
-		if uint64(bm.Maximum()) < from {
-			break
-		}
-		noReasonToCheckNextShard := uint64(bm.Minimum()) <= from && uint64(bm.Maximum()) >= to
+		noReasonToCheckNextShard := (uint64(bm.Minimum()) <= from && uint64(bm.Maximum()) >= to) || binary.BigEndian.Uint32(k[len(k)-4:]) == ^uint32(0)
 
-		updated++
 		bm.RemoveRange(from, to)
 		if bm.GetCardinality() == 0 { // don't store empty bitmaps
 			err = c.DeleteCurrent()
```

Comment:
Is this a dangerous operation?

Reply:
I better will use dedicated cursor here and later switch to ETL. 
But if I understand assertion problem correctly - it happens when you iterate over dupsort bucket and delete last duplicated value (or do something else what causing conversion page from leaf to non-leaf). But I hadn't reproduce it. Maybe need to do something like fuzz testing to reproduce it (randomly add/delete/iterate - but use what we know about IH bucket - values have different length, custom comparator used, bucket dupsorted).